### PR TITLE
Support newInstanceNamed with empty name

### PR DESCRIPTION
**Pull request by mattrberry**
Opts to rewrite  `refer('Foo').newInstanceNamed('', [])` as `Foo` rather than `Foo.`. Two primary reasons
1. `Foo.` breaks compilation with bad error messages
2. [Analyzer's `ConstructorElement#name`](https://github.com/dart-lang/sdk/blob/e8939b0ff089a97297cf28626be46c3c75fe72e7/pkg/analyzer/lib/dart/element/element.dart#L466) is a non-nullable String, which pushes the `isEmpty` check to user code

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

### CODE REVIEW

**Review by srawlins — APPROVED** [review]
Thanks! ❤️ 

I'm not a primary owner so I think we should wait for one more review to land.

**Comment by mosuem** [comment]
Closing as the [dart-lang/code_builder](https://github.com/dart-lang/code_builder) repository is merged into the [dart-lang/tools](https://github.com/dart-lang/tools) monorepo. Please re-open this PR there!

**Pull request closed without merging**
